lib/gui/components/modal/services/modal.js

```diff
@@ -56,18 +56,32 @@ module.exports = function($uibModal, $q) {
       close: modal.close,
       result: $q((resolve, reject) => {
         modal.result
-          .then(resolve)
-          .catch((error) => {
+          .then((value) => {
+            AnalyticsService.logEvent('Modal closed', {
+              method: 'User accepted',
+              value: value
+            });
 
+            resolve(value);
+          })
+          .catch((error) => {
             // Bootstrap doesn't 'resolve' these but cancels the dialog;
             // therefore call 'resolve' here applied to 'false'.
             if (error === 'escape key press' || error === 'backdrop click') {
+              AnalyticsService.logEvent('Modal closed', {
+                method: error
+              });
+
               resolve();
 
             // For some annoying reason, UI Bootstrap Modal rejects
```

Comment:
After the edit to the `else if`, this comment no longer makes sense here, but should be moved to the top `if`.